compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -599,8 +599,9 @@ impl<'f> PerFunctionContext<'f> {
     }
 
     fn update_data_bus(&mut self) {
-        let databus = self.inserter.function.dfg.data_bus.clone();
-        self.inserter.function.dfg.data_bus = databus.map_values(|t| self.inserter.resolve(t));
+        let mut databus = self.inserter.function.dfg.data_bus.clone();
```

Comment:
We can avoid this clone, no?

Reply:
No, `self.inserter.resolve(t)` borrows inserter mutably so we can't also access `self.inserter.function.dfg.data_bus`

Reply:
Ah that's unfortunate.